src/distilabel/pipeline/templates/instruction.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from distilabel.distiset import Distiset
+from distilabel.llms.base import LLM
+from distilabel.llms.huggingface import InferenceEndpointsLLM
+from distilabel.pipeline import Pipeline
+from distilabel.steps import KeepColumns
+from distilabel.steps.tasks import MagpieGenerator
+
+MODEL = "meta-llama/Meta-Llama-3.1-8B-Instruct"
+
+
+class InstructionResponsePipeline:
+    """Generates instructions and responses for a given system prompt.
+
+    This example pipeline can be used for a Supervised Fine-Tuning dataset which you
+    could use to train or evaluate a model. The pipeline generates instructions using the
+    MagpieGenerator and responses for a given system prompt. The pipeline then keeps only
+    the instruction, response, and model_name columns.
+
+    References:
+        - [Magpie: Alignment Data Synthesis from Scratch by Prompting Aligned LLMs with Nothing](https://arxiv.org/abs/2406.08464)
+
+    Example:
+
+        Generate instructions and responses for a given system prompt:
+
+        ```python
+        from distilabel.pipeline import InstructionResponsePipeline
+
+        pipeline = InstructionResponsePipeline()
+
+        distiset = pipeline.run()
```

Comment:
Have you tested this runs without the 
```python
if __name__ == "__main__":
    pipeline.run()
```
outside of a notebook? To mention that in the docs

Reply:
I've tested it in a notebook and a script. The ideal solution is that the class can be imported into notebooks and scripts, but `instruction.py` does not need to be run as a script.